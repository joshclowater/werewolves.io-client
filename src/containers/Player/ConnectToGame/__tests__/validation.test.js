import validate from '../validation';

describe('with valid values', () => {
  let values;
  beforeEach(() => {
    values = {
      gameId: '12345',
      name: 'mock name'
    };
  });
  it('returns empty object', () => {
    expect(validate(values)).toEqual({});
  });
});

describe('with no values', () => {
  let values;
  beforeEach(() => {
    values = {};
  });
  it('returns non empty errors', () => {
    expect(validate(values)).toMatchSnapshot();
  });
});

describe('with invalid length', () => {
  let values;
  beforeEach(() => {
    values = {
      gameId: '1234',
      name: '1234567890123'
    };
  });
  it('returns invalid length errors', () => {
    expect(validate(values)).toMatchSnapshot();
  });
});
