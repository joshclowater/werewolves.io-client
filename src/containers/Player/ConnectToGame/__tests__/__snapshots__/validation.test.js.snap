// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with invalid length returns invalid length errors 1`] = `
Object {
  "gameId": "Game id must be 5 characters",
  "name": "Name must be less than 12 characters",
}
`;

exports[`with no values returns non empty errors 1`] = `
Object {
  "gameId": "Please enter a game id",
  "name": "Please enter a name",
}
`;
